dora/core/server/master/src/main/java/alluxio/master/job/DoraLoadJob.java

```diff
@@ -85,47 +96,129 @@
 public class DoraLoadJob extends AbstractJob<DoraLoadJob.DoraLoadTask> {
   private static final Logger LOG = LoggerFactory.getLogger(DoraLoadJob.class);
   public static final String TYPE = "load";
-  private static final int RETRY_BLOCK_CAPACITY = 1000;
-  private static final double RETRY_THRESHOLD = 0.8 * RETRY_BLOCK_CAPACITY;
   private static final int BATCH_SIZE = Configuration.getInt(PropertyKey.JOB_BATCH_SIZE);
 
   // Job configurations
   private final String mLoadPath;
-  private OptionalLong mBandwidth;
-  private boolean mUsePartialListing;
-  private boolean mVerificationEnabled;
+  private final OptionalLong mBandwidth;
+  private final boolean mUsePartialListing;
+  private final boolean mVerificationEnabled;
 
   // Job states
-  private final Queue<LoadSubTask> mRetrySubTasks = new ArrayDeque<>();
-  private final Map<LoadSubTask, Integer> mRetryCount = new ConcurrentHashMap<>();
-  private final Map<String, String> mFailedFiles = new HashMap<>();
+  private final Queue<LoadSubTask> mRetrySubTasksDLQ = new ArrayDeque<>();
+  // Only the most recent 1k failures are persisted.
+  // If more are needed, can turn on debug LOG for this class.
+  private final Queue<Pair<LoadSubTask, String>> mRecentFailures =
```

Comment:
Previously we store the whole error messages for every error. Each entry takes up 2KB memory and I think it's too much. Plus --verbose cannot really display these much information. So I only keep the most recent 1000 ones to avoid OOM.